### DISCUSSION TITLE
[LOGS] - Add info to http access log

### DISCUSSION THREAD

**Discussion by FernandoIberian** [roadrunner-server/roadrunner#1904] [Q&A] [answered]
Hi! Is there any way i can shove some information onto the http access logs?

Every request is logged like so:
`http access log	{"read_bytes": 0, "write_bytes": 46446, "status": 200, "method": "GET", "URI": "", "remote_address": "A", "query": "", "content_len": 0, "host": "host", "user_agent": "", "referer": "referer", "time_local": "23/Apr/24:12:56:31 +0000", "request_time": "2024-04-23T12:56:31+0000", "start": "2024-04-23T12:56:30+0000", "elapsed": 89}`

And i want to add USERID:
`http access log	{"read_bytes": 0, "write_bytes": 46446, "status": 200, "method": "GET", "URI": "", "remote_address": "A", "query": "", "content_len": 0, "host": "host", "user_agent": "", "referer": "referer", "time_local": "23/Apr/24:12:56:31 +0000", "request_time": "2024-04-23T12:56:31+0000", "start": "2024-04-23T12:56:30+0000", "elapsed": 89, **"USERID":"1"**}`

Thank you.

**Comment by rustatian** [comment]
Hey @FernandoIberian 👋 
Do you mean system userid (`uid`)?

**Reply by FernandoIberian** [reply]
I've explained myself wrong, what i want to achieve is identify each request with it's corresponding user. In my case, every request has a token (JWT) attached to the "Authorization" header. The idea would be log that info (not the entire JWT, just the payload) on the [http access log](https://roadrunner.dev/docs/lab-access-logs/current/en) in order to identify WHO made the request.

I can easily do it if i register an [AppLoger](https://roadrunner.dev/docs/lab-applogger/current/en), but since roadrunner-server log [already does it](https://github.com/roadrunner-server/http/blob/master/middleware/log.go#L142) (elapsed time, URI, query, remote_addr...), im wondering if there is any way i can just shove some extra info on that log.

Thank you for the quick follow up!

**Reply by rustatian** [reply, marked as answer]
Ah, got u, thanks for the explanation. Unfortunately, due to security reasons, RR logs only well-known fields, w/o logging the entire Headers for example.
If you need to log some fields, specific to your use-case - it is better to use AppLogger or simply push a message to the STDERR (would be logger at INFO log level).
